### ISSUE TITLE
ERROR: Assertion 'all_nonnegative([chamfer])' failed: "Chamfer must be a nonnegative number"

### ISSUE THREAD

**Issue by argoolsbee** [jyap808/OpenSCADModels#1]
Running the script with no changes returned the following error:

> ERROR: Assertion 'all_nonnegative([chamfer])' failed: "Chamfer must be a nonnegative number" in file ../Documents/OpenSCAD/libraries/BOSL2/masks3d.scad, line 148

Changing line 32 and 37 to `chamfer = chamfer_mm` resolved this for me.

**Comment by jyap808** [comment]
Hmm, I only see this error if chamfer mm is set to a negative number.

The default is `chamfer_mm = 2;`

![image](https://github.com/jyap808/OpenSCADModels/assets/710537/2bd1be75-84ca-48d8-99d9-a0783c2bcfbf)

**Comment by argoolsbee** [comment]
It happens for me no matter what chamfer_mm is set to. This is from opening the original file:

![image](https://github.com/jyap808/OpenSCADModels/assets/3212135/5dea4e69-5e90-4f58-8ecc-25da25889d5f)

I'm on OpenSCAD version 2021.01, on a Mac. I also made sure I have the latest BOSL2 libraries.

**Comment by argoolsbee** [comment]
Putting some echo's in chamfered_cylinder confirms that chamfer is not getting passed for me.

![image](https://github.com/jyap808/OpenSCADModels/assets/3212135/36f6b976-b64d-423c-a381-d4e43e90a2ad)

Removing `h=` and `d=` when calling chamfered_cylinder also resolves the error, so it seems something about my config does not like the mismatched argument passing styles.

```
// Create the head
module head() {
    chamfered_cylinder(dial_height + head_height + chamfer_mm, spool_width + 0.5, chamfer_mm);
}

// Create the dial
module dial() {
    chamfered_cylinder(dial_height, (2 * brim_width) + spool_width, chamfer_mm);
}
```

**Comment by jyap808** [comment]
OK, that's interesting. I'm actually using OpenSCAD built from source on my Mac. I suspect it's a bug in that version that's been fixed.

**Comment by argoolsbee** [comment]
I also confirmed the argument passing is fixed but not released yet by using the latest dev snapshot.

**Comment by jyap808** [comment]
Tested with OpenSCAD 2021.01 and fixed in [951386fffc938a02f53e65205e0bd13e718ddccf](https://github.com/jyap808/OpenSCADModels/commit/951386fffc938a02f53e65205e0bd13e718ddccf)

Thanks for the report @argoolsbee !
